### PR TITLE
feat: nit extraction + compiler gate + CI wind tunnel

### LINKED ISSUE #821: fix: totem extract should parse CodeRabbit review body nits, not just inline comments
totem extract only fetches PR inline review comments via the GitHub API. CodeRabbit posts nits in a collapsed section of the main review body comment, not as inline comments. These nits are currently invisible to the extraction pipeline. Update extract to also parse the review body for collapsed nit sections.

### PR DESCRIPTION
**Pull request by satur8d**
## Summary
- Extract CodeRabbit nits from collapsed review body sections (#821)
- Downgrade untested Pipeline 1 error rules to warning during compilation (strategy#59)
- Add wind tunnel (totem test) to CI pipeline (strategy#58)
- Fix shield findings: ESM require crash, null author safety, global regex state

Closes #821

## Test plan
- [x] `pnpm run format:check` — pass
- [x] `totem lint` — 219 rules, 0 violations
- [x] `totem test` — 9/9 fixtures pass
- [x] `totem shield` — reviewed, findings fixed
- [x] Parse nits tests — 10/10 pass
- [x] Extract tests — 71/71 pass
- [x] Full CLI tests — 646/646 pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Compilation now downgrades rule severity to warning for lessons not covered by detected tests.
  * Extraction now incorporates CodeRabbit "nit" comments into PR analysis to surface actionable nits.

* **Tests**
  * Added comprehensive tests validating nit parsing behavior, edge cases, and performance.

* **Chores**
  * CI pipeline now runs an additional test invocation as part of the build.

### CODE REVIEW

File: packages/cli/src/parse-nits.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      depth--;
+      if (depth === 0) {
+        return body.slice(startPos, closeMatch.index);
+      }
+      pos = closeMatch.index + closeMatch[0].length;
+    }
+  }
+
+  return null;
+}
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/parse-nits.ts]: Intentional — Shield flagged global regex state mutation as unsafe. Creating fresh regexes per iteration avoids lastIndex corruption. The performance cost is negligible for the input sizes we process (PR review bodies).
